src/nupic/types/Sdr.cpp

```diff
@@ -135,6 +135,24 @@ namespace sdr {
     SparseDistributedRepresentation::~SparseDistributedRepresentation()
         { deconstruct(); }
 
+
+    void SparseDistributedRepresentation::reshape(const vector<UInt> &dimensions) const {
+        // Make sure we have the data in a format which does not care about the
+        // dimensions, IE: dense or sparse but not coordinates
+        if( not dense_valid and not sparse_valid )
+            getSparse();
+        coordinates_valid = false;
+        coordinates_.assign( dimensions.size(), {} );
+        dimensions_ = dimensions;
+        const auto oldSize = size;
+        // Calculate the SDR's size.
+        UInt newSize = 1;
+        for(UInt dim : dimensions)
+            newSize *= dim;
```

Comment:
nit, please consider replacing this chunk of code with a one liner: 
` int product = std::accumulate(v.begin(), v.end(), 1, std::multiplies<int>());`